frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1160,10 +1165,58 @@ function ReaderFooter:addToMainMenu(menu_items)
                         callback = function()
                             self.settings.toc_markers = not self.settings.toc_markers
                             self:setTocMarkers()
-                            self:updateFooter()
-                            UIManager:setDirty(nil, "ui")
+                            self:refreshFooter(true)
                         end
                     },
+                    {
+                        text_func = function()
+                            local markers_width_text = _("thick")
+                            if self.settings.toc_markers_width == 1 then
+                                markers_width_text = _("thin")
+                            elseif self.settings.toc_markers_width == 2 then
+                                markers_width_text = _("normal")
+                            end
+                            return T(_("Chapter markers width (%1)"), markers_width_text)
+                        end,
+                        enabled_func = function()
+                            return not self.settings.progress_style_thin and self.settings.toc_markers
+                        end,
+                        sub_item_table = {
+                            {
+                                text = _("Thin"),
+                                checked_func = function()
+                                    return self.settings.toc_markers_width == 1
+                                end,
+                                callback = function()
+                                    self.settings.toc_markers_width = Screen:scaleBySize(1)
```

Comment:
Looks like this might work on an emulator where scaleBySize does x1 - but unless I miss something, it won't be checked (or worse) when scaleBySize does x1.7 or x2.
So, may be just save the unscale value - and do the scaleBySize on lines 1330 and/or 338.